code/game/objects/items/stacks/sheets/sheet_types.dm

```diff
@@ -57,6 +57,20 @@ var/global/list/datum/stack_recipe/metal_recipes = list ( \
 	flags = CONDUCT
 	origin_tech = "materials=1"
 
+/obj/item/stack/sheet/metal/attackby(obj/item/weapon/W, mob/living/user)
+	..()
+	if(istype(W, /obj/item/weapon/paper/talisman/construction))
```

Comment:
Oh right; I should note that accessing the metal's vars if using afterattack requires you do target.var/proc/whatever or it'll assume you're accessing the src's vars.